lapis2-docs/src/content/docs/architecture-and-dev-docs/architecture-and-constraints.mdx

```diff
@@ -0,0 +1,23 @@
+---
+title: Architecture and Constraints
+description: Overview of the architecture and constraints of the software.
+---
+
+We identified the following constraints for our software:
+
+- Developed under an open-source licence. We chose the tooling such that a broad spectrum of developers can in principle
+work on the software.
+- The software is designed to be highly configurable so that it can be used for various organisms.
+Configuration files have to be passed to LAPIS and SILO at runtime that determine the nature of the organism such as:
+  - a reference genome
+  - which metadata is available on the genomic data
+- Input files for SILO are provided in a specific format:
+  - metadata and quality control
+  - sequence data (non-aligned nucleotide sequences, aligned nucleotide sequences, aligned AA sequences)
+- LAPIS is backed by a database that understands SILO queries.
+- The system is designed to have the best possible performance.
+This mostly targets SILO, but also in LAPIS,
+we have to keep in mind that we are dealing with potentially large data that we have to serve to the client.
+- We provide executable that can be run on Linux and macOS.
```

Comment:
Do we provide them?

Reply:
Actually not yet. Should we simply remove this sentence?